### PR TITLE
Execute same requests only once, all other requests wait

### PR DESCRIPTION
**Pull request by ChaituVR**
Fixes snapshot-labs/snapshot#1203

### CODE REVIEW

**Comment by ChaituVR** [comment]
Comments from Fabien:
- Reuse the key
- Move the logic from middleware to https://github.com/snapshot-labs/snapshot-score/blob/master/src/scores.ts#L29 
- Make it as simple as possible

File: src/scores.ts [hunk 1]
```diff
@@ -60,3 +56,21 @@ export default async function scores(parent, args) {
...
+export default async function scores(parent, args) {
+  const key = createHash('sha256')
+    .update(JSON.stringify(args))
+    .digest('hex');
+  // console.log('Key', key, JSON.stringify({ space, strategies, network }), addresses.length);
+
+  return new Promise(async resolve => {
+    // Wait for scores to be calculated
+    eventEmitter.once(key, data => resolve(data));
+    console.log(eventEmitter.listenerCount(key));
```
Comment (ChaituVR) [inline src/scores.ts:69]:
```suggestion
```

**Review by mktcode — APPROVED** [review]
Yeah let's do caching but... without a cache. How elegant is that?!!?!? Love it!

File: src/scores.ts [hunk 2]
```diff
@@ -23,6 +23,14 @@ async function getBlockNum(network) {
 
 async function calculateScores(args, key) {
   const { space = '', strategies, network, addresses } = args;
+
+  const key = createHash('sha256')
```
Comment (mktcode) [inline src/scores.ts:27]: @bonustrack @ChaituVR This last merge commit messed things up a bit. This here won't work. (`key` already declared)

Comment (ChaituVR) [reply src/scores.ts]: Fixed it

**Pull request merged**